src/plugins/intel_gna/frontend/model_quantizer.hpp

```diff
@@ -22,125 +22,125 @@
+        return quantize(model, [](InferenceEngine::CNNNetwork&, bool run_before_copy, bool low_precision){}, scale_factors);
     }
 
     template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model,  const PreQuantisationCb &cb, std::vector<float> scaleFactors) const {
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,  const PreQuantisationCb& cb, std::vector<float> scale_factors) const {
         GNAPluginNS::GnaInputs inputs;
         InferenceEngine::InputsDataMap inputs_map = model.getInputsInfo();
         int sf_id = 0;
         for (auto &&input_data : inputs_map) {
             auto input_layer = getCreatorLayer(input_data.second->getInputData()).lock();
-            if (scaleFactors.size() <= sf_id) {
+            if (scale_factors.size() <= sf_id) {
                 THROW_GNA_EXCEPTION << "Scale factors are not set for some of the inputs";
             }
-            inputs[input_layer->name].scale_factor = scaleFactors[sf_id++];
+            inputs[input_layer->name].scale_factor = scale_factors[sf_id++];
         }
 
-        return quantize(model, cb, inputs);
+        return quantize(model, cb, inputs, InferenceEngine::Precision::I16, false);
     }
 
+    // This function is acutally used by the plugin
     template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork &model,
-                                         const PreQuantisationCb &cb,
-                                         const GNAPluginNS::GnaInputs &inputs) const {
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,
+                                         const PreQuantisationCb& cb,
+                                         const GNAPluginNS::GnaInputs& inputs,
+                                         const InferenceEngine::Precision& gna_precision,
+                                         const bool& inputs_int8_precision) const {
         OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "ModelQuantizer::quantize");
         auto visitor = [&](InferenceEngine::CNNLayerPtr lp) {
-            auto newLayer = InferenceEngine::injectData<QuantizedLayerParams>(lp);
-            transformLayer(newLayer, WeightsConverter());
-            return newLayer;
+            auto new_layer = InferenceEngine::injectData<QuantizedLayerParams>(lp);
+            transformLayer(new_layer, WeightsConverter());
+            return new_layer;
         };
-        bool lowPrecision = (T::mandatory().getInputPrecision().size() == sizeof(uint8_t));
-        InferenceEngine::CNNNetwork copiedNet = InferenceEngine::CNNNetCopy(model);
-        cb(copiedNet, true, lowPrecision);
 
-        copiedNet = InferenceEngine::CNNNetCopy(copiedNet, visitor);
+        // User's hints for GNA weights precision
+        const bool weights_int8_precision = (gna_precision == InferenceEngine::Precision::I8) ? true : false;
```

Comment:
I'll leave it for now as it is, because inputs_int8_precision is most probably going to be removed soon. Only changed variable names to use "int8" instead of "low".